palette/transaction-payment/src/lib.rs

```diff
@@ -154,28 +154,28 @@ impl<T: Trait + Send + Sync> ChargeTransactionPayment<T> {
 	where
 		BalanceOf<T>: Sync + Send,
 	{
-		let len_fee = if info.pay_length_fee() {
+		if info.pays_fee {
 			let len = <BalanceOf<T>>::from(len);
 			let base = T::TransactionBaseFee::get();
 			let per_byte = T::TransactionByteFee::get();
-			base.saturating_add(per_byte.saturating_mul(len))
-		} else {
-			Zero::zero()
-		};
+			let len_fee = base.saturating_add(per_byte.saturating_mul(len));
 
-		let weight_fee = {
-			// cap the weight to the maximum defined in runtime, otherwise it will be the `Bounded`
-			// maximum of its data type, which is not desired.
-			let capped_weight = info.weight.min(<T as system::Trait>::MaximumBlockWeight::get());
-			T::WeightToFee::convert(capped_weight)
-		};
+			let weight_fee = {
```

Comment:
hmm I think now we want to have any fee only when `pays_fee == true`? then weight fee should also go inside the `if`, no?

Reply:
this is already inside the if. It can still pay tips even pays_fee is false, I guess that make sense.

Reply:
ahh yeas indeed, i did not see the indentation. Kinda hard to check this stuff in gh ui :D